clang/lib/Sema/CheckedCAnalysesPrepass.cpp

```diff
@@ -23,27 +23,38 @@ class PrepassHelper : public RecursiveASTVisitor<PrepassHelper> {
   private:
     Sema &SemaRef;
     PrepassInfo &Info;
-    // InBoundsExpr tracks whether the expressions being visited are
-    // within a declared bounds expression.
-    bool InBoundsExpr = false;
+
+    // VarWithBounds is a variable that has a bounds expression. It is used to
+    // keep track of the following:
+    // 1. Whether the expressions being visited are within a declared bounds
+    // expression.
```

Comment:
Does `VarWithBounds` also track:
1. Whether the expressions being visited are within a bounds expression declared by a where clause, and:
2. The variable with which a declared bounds expression is associated?

Maybe this comment could be reworded to reflect the fact that `VarWithBounds` always tracks the variable with which a declared or where clause bounds expression is associated, and the fact that if `VarWithBounds` is non-null, a visited expression is within a declared or where clause bounds expression?